internal/infra/repository/author_repository_test.go

```diff
@@ -0,0 +1,141 @@
+	req := testcontainers.ContainerRequest{
+		Image:        "postgres:15",
+		Env:          map[string]string{"POSTGRES_USER": "postgres", "POSTGRES_PASSWORD": "postgres", "POSTGRES_DB": "lucienne_test"},
+		ExposedPorts: []string{"5432/tcp"},
+		WaitingFor:   wait.ForLog("database system is ready to accept connections").WithOccurrence(2).WithStartupTimeout(10 * time.Second),
+	}
+	container, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	})
+	require.NoError(t, err, "Falha ao iniciar o container do banco de dados")
+
+	// Obtém o host e a porta mapeada dinamicamente do container.
+	endpoint, err := container.Endpoint(ctx, "")
+	require.NoError(t, err, "Falha ao obter o endpoint do container")
+
+	// Constrói a URL de conexão completa usando o endpoint dinâmico.
+	databaseURL := fmt.Sprintf("postgres://postgres:postgres@%s/lucienne_test?sslmode=disable", endpoint)
+
+	// AGORA, com o container rodando e a URL dinâmica em mãos, nós configuramos e conectamos ao banco.
+	config.EnvVariables.DatabaseURL = databaseURL
+	database.ConnectDB()
+
+	// Caminho para as migrações relativo ao arquivo de teste
+	migrationsPath := "file://../../../db/migrations"
+
+	// Usa a URL dinâmica para as migrações.
+	m, err := migrate.New(migrationsPath, databaseURL)
+	require.NoError(t, err, "Falha ao criar instância de migração")
+
+	err = m.Up()
+	if err != nil && err != migrate.ErrNoChange {
+		require.NoError(t, err, "Falha ao aplicar migrações (Up)")
+	}
+
+	// Retorna a função de limpeza (teardown)
+	return func() {
+		err = m.Down()
+		if err != nil && err != migrate.ErrNoChange {
+			log.Printf("AVISO: Falha ao reverter migrações (Down): %v", err)
+		}
+		sourceErr, dbErr := m.Close()
+		require.NoError(t, sourceErr)
+		require.NoError(t, dbErr)
+		database.Conn.Close(context.Background())
+
+		if err := container.Terminate(ctx); err != nil {
+			t.Fatalf("failed to terminate container: %s", err)
+		}
+	}
+}
+
+func TestPostgresAuthorRepository_UpdateAuthor(t *testing.T) {
+	// Só executa este teste se a flag -short não for usada
+	if testing.Short() {
+		t.Skip("Pulando teste de integração com banco de dados em modo 'short'.")
+	}
+
+	cleanup := setupTestDBAndMigrate(t)
+	defer cleanup()
+	ctx := context.Background()
+	repo := repository.NewPostgresAuthorRepository()
+
+	// -- Caso de Sucesso --
+	t.Run("deve atualizar um autor com sucesso", func(t *testing.T) {
+		// 1. Inserir um autor para o teste
+		var authorID int
+		originalName := "Autor Original Para Teste"
+		err := database.Conn.QueryRow(ctx, insertQuery, originalName).Scan(&authorID)
+		require.NoError(t, err, "Falha ao inserir autor para o teste de atualização")
+
+		// Garante que o autor de teste seja removido no final
+		defer func() {
+			_, err := database.Conn.Exec(ctx, deleteQuery, authorID)
+			if err != nil {
+				log.Printf("AVISO: Falha ao limpar autor de teste com ID %d: %v", authorID, err)
```

Comment:
```suggestion
				t.Logf("AVISO: Falha ao limpar autor de teste com ID %d: %v", authorID, err)
```

Mesma coisa do caso anterior.